test/Select.js

```diff
@@ -130,6 +130,51 @@ describe('Select', function() {
 		assert.ok(!dropdown.expanded);
 	});
 
+	it('should not open dropdown when select component is disabled', function() {
+		select = new Select({
+			disabled: true,
+			items: ['First', 'Second', 'Third']
+		});
+
+		var dropdown = select.getDropdown();
+		assert.ok(!dropdown.expanded);
+
+		dom.triggerEvent(select.element.querySelector('button'), 'click');
+		assert.ok(!dropdown.expanded);
+	});
+
+	it('should close dropdown when select component gets disabled', function(done) {
+		select = new Select({
+			items: ['First', 'Second', 'Third']
+		});
+
+		dom.triggerEvent(select.element.querySelector('button'), 'click');
+
+		var dropdown = select.getDropdown();
+		assert.ok(dropdown.expanded);
+		
+		select.disabled = true;
+		select.once('stateChanged', function() {
+			assert.ok(!dropdown.expanded);
+			done();
+		});
+	});
+
+	it('should `selectedIndex` get change even select component is disabled', function(done) {
```

Comment:
According to a real `<select>`, this is totally possible. Once the `disable` state is only for blocking users interactions. For example, it is possible changing the value of a disabled `<select>` by using `document.querySelector('select').value = 'on of the options'`.